src/Api/FingerprintApi.php

```diff
@@ -1,739 +1,142 @@
- *
- * OpenAPI spec version: 3
- * Contact: support@fingerprint.com
- * Generated by: https://github.com/swagger-api/swagger-codegen.git
- * Swagger Codegen version: 3.0.34
- */
-/**
- * NOTE: This class is auto generated by the swagger code generator program.
- * https://github.com/swagger-api/swagger-codegen
- * Do not edit the class manually.
- */
 
 namespace Fingerprint\ServerAPI\Api;
 
+use Fingerprint\ServerAPI\Model\EventResponse;
+use Fingerprint\ServerAPI\Model\Response;
+use Fingerprint\ServerAPI\ObjectSerializer;
 use GuzzleHttp\Client;
 use GuzzleHttp\ClientInterface;
-use GuzzleHttp\Exception\RequestException;
-use GuzzleHttp\Psr7\MultipartStream;
+use Fingerprint\ServerAPI\Configuration;
 use GuzzleHttp\Psr7\Request;
 use GuzzleHttp\RequestOptions;
-use Fingerprint\ServerAPI\ApiException;
-use Fingerprint\ServerAPI\Configuration;
-use Fingerprint\ServerAPI\HeaderSelector;
-use Fingerprint\ServerAPI\ObjectSerializer;
+use RuntimeException;
 
-/**
- * FingerprintApi Class Doc Comment
- *
- * @category Class
- * @package  Fingerprint\ServerAPI
- * @author   Swagger Codegen team
- * @link     https://github.com/swagger-api/swagger-codegen
- */
 class FingerprintApi
 {
     /**
      * @var ClientInterface
      */
-    protected $client;
-
+    protected ClientInterface $client;
     /**
      * @var Configuration
      */
-    protected $config;
-
-    /**
-     * @var HeaderSelector
-     */
-    protected $headerSelector;
+    protected Configuration $config;
 
-    protected $integration_info = 'fingerprint-pro-server-php-sdk/4.1.0';
+    protected string $integration_info = 'fingerprint-pro-server-php-sdk/{version}';
 
     /**
-     * @param ClientInterface $client
-     * @param Configuration   $config
-     * @param HeaderSelector  $selector
+     * @param ClientInterface|null $client
+     * @param Configuration|null $config
      */
     public function __construct(
         ClientInterface $client = null,
-        Configuration $config = null,
-        HeaderSelector $selector = null
-    ) {
+        Configuration   $config = null
+    )
+    {
         $this->client = $client ?: new Client();
         $this->config = $config ?: new Configuration();
-        $this->headerSelector = $selector ?: new HeaderSelector();
-
-        // TODO Remove on next major release
-        if(version_compare(
-            PHP_VERSION,
-            '8.1',
-            '<'
-        )) {
-            trigger_error('You are using a PHP version that has reached EOL. Support for it will be removed in next major release of this SDK. Please consider upgrading to PHP 8.1 or higher.', E_USER_WARNING);
-        }
     }
 
-    /**
-     * @return Configuration
-     */
-    public function getConfig()
+    protected function getSDKVersion(): string
     {
-        return $this->config;
-    }
+        $string = file_get_contents(dirname(__FILE__) . "/../../composer.json");
+        $jsonData = json_decode($string, true);
```

Comment:
This is going to run on every request, right? We should consider reading this file in constructor and storing version in a class property instead.

Also, consider checking if [this](https://getcomposer.org/doc/07-runtime.md#installed-versions) API is available and if it is, pls use it instead.